### PR TITLE
Minor improvements to Param docs

### PR DESCRIPTION
**Pull request by fbertsch**
- Add `print_it` function which helps clarify example (previously, the `print_it` was used before an actual `print` lambda. This condenses both into a `print_it` and defines that function before it is used)
- Change first instance of `int_param` to `x` to match previous param examples (previously, `int_param` was used in a task before it was ever introduced as a new param)

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: docs/apache-airflow/core-concepts/params.rst [hunk 1]
```diff
@@ -90,7 +90,7 @@ You can change this by setting ``render_template_as_native_obj=True`` while init
 
     with DAG(
         "the_dag",
-        params={"x": Param(5, type="integer", minimum=3)},
+        params={"int_param": Param(5, type="integer", minimum=3)},
         render_template_as_native_obj=True
```
Comment (fbertsch) [inline docs/apache-airflow/core-concepts/params.rst]:
```suggestion
        render_template_as_native_obj=True,
```

**Review by uranusjr — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
doc build is failing - I recomend to run `breeze build-docs` before rebasing.

**Comment by eladkal** [comment]
@fbertsch can you fix the errors?

**Comment by eladkal** [comment]
fixed in https://github.com/apache/airflow/pull/30345

**Pull request closed without merging**